ckb-bin/src/subcommand/reset_data.rs

```diff
@@ -0,0 +1,93 @@
+use std::fs;
+use std::path::PathBuf;
+
+use std::io::{stdin, stdout, Write};
+
+use ckb_app_config::{ExitCode, ResetDataArgs};
+
+fn join_paths(paths: Vec<PathBuf>) -> String {
+    paths
+        .iter()
+        .map(|pb| pb.clone().into_os_string().into_string().unwrap())
+        .collect::<Vec<String>>()
+        .join(", ")
+}
+
+pub fn reset_data(args: ResetDataArgs) -> Result<(), ExitCode> {
+    let stdout = stdout();
+    let mut stdout = stdout.lock();
+    let stdin = stdin();
+
+    let mut target_dirs = vec![];
+    let mut target_files = vec![];
+
+    if args.all {
+        target_dirs.push(args.data_dir);
+    }
+
+    if args.database {
+        target_dirs.push(args.db_path);
+    }
+
+    if args.database || args.indexer {
+        target_dirs.push(args.indexer_db_path);
+    }
+
+    if args.network {
+        target_dirs.push(args.network_dir);
+    }
+
+    if args.network_peer_store {
+        target_files.push(args.network_peer_store_path);
+    }
+
+    if args.network_secret_key {
+        target_files.push(args.network_secret_key_path);
+    }
+
+    if args.logs {
+        if let Some(dir) = args.logs_dir {
+            target_dirs.push(dir);
+        }
+    }
+
+    if !args.force {
+        let mut prompt = String::from("remove ");
+        prompt.push_str(&join_paths(target_dirs.clone()));
```

Comment:
I don't think we need to clone `target_dirs` and `target_files` here, we can use `chain` and `to_string_lossy` to simplify the code:
```suggestion
    let to_be_deleted_targets_str = target_dirs.iter()
            .chain(target_files.iter())
            .map(|path| path.to_string_lossy())
            .collect::<Vec<_>>()
            .join(", ");
    
    let input = prompt(format!("remove {}?", to_be_deleted_targets_str));
```

Reply:
sure